torch/distributed/_tensor/placement_types.py

```diff
@@ -182,27 +227,44 @@ def _to_replicate_tensor(
         """
         my_coordinate = mesh.get_coordinate()
         num_chunks = mesh.size(dim=mesh_dim)
+
         # TODO: what should happen if rank is not in the mesh?
         # see issue https://github.com/pytorch/tau/pull/492
         assert (
             my_coordinate is not None
         ), "Rank if not part of mesh"  # TODO: figure out behavior here
         # check if it needs to pad input tensor before all_gather
-        pad_idx = size[self.dim] % num_chunks
-        if pad_idx != 0 and my_coordinate[mesh_dim] >= pad_idx:
-            local_tensor = self._pad_tensor(local_tensor)
+        full_chunk_size = (size[self.dim] + num_chunks - 1) // num_chunks
+        chunk_sizes = [
+            max(
+                min(size[self.dim], full_chunk_size * (idx + 1))
+                - full_chunk_size * idx,
+                0,
+            )
+            for idx in range(num_chunks)
+        ]
+        pad_sizes = [full_chunk_size - chunk_size for chunk_size in chunk_sizes]
+        is_padded = not all(pad_size == 0 for pad_size in pad_sizes)
```

Comment:
nit: why don't we use a similar condition for `is_padded` in `reduce_scatter_tensor`? i.e. `size[self.dim]% num_chunks != 0`